### PR TITLE
Add DualStack tests to psm_interop_kokoro_lib.sh

### PR DESCRIPTION
**Pull request by aranjans**
Added dualstack test to  `psm_interop_kokoro_lib.sh`.

### CODE REVIEW

File: .kokoro/psm_interop_kokoro_lib.sh [hunk 1]
```diff
@@ -694,6 +739,10 @@ activate_gke_cluster() {
       GKE_CLUSTER_NAME="interop-test-psm-basic"
       GKE_CLUSTER_ZONE="us-central1-c"
       ;;
+    GKE_CLUSTER_DUAL_STACK)
+          GKE_CLUSTER_NAME="psm-interop-lb-primary-ds"
+          GKE_CLUSTER_ZONE="us-central1-c"
```
Comment (purnesh42H) [inline .kokoro/psm_interop_kokoro_lib.sh:744]: zone should us-central1-a

File: .kokoro/psm_interop_kokoro_lib.sh [hunk 2]
```diff
@@ -694,6 +739,10 @@ activate_gke_cluster() {
       GKE_CLUSTER_NAME="interop-test-psm-basic"
       GKE_CLUSTER_ZONE="us-central1-c"
       ;;
+    GKE_CLUSTER_DUAL_STACK)
+          GKE_CLUSTER_NAME="psm-interop-lb-primary-ds"
+          GKE_CLUSTER_ZONE="us-central1-a"
+          ;;
```
Comment (sergiitk) [inline .kokoro/psm_interop_kokoro_lib.sh:745]:
```suggestion
    GKE_CLUSTER_DUAL_STACK)
      GKE_CLUSTER_NAME="psm-interop-lb-primary-ds"
      GKE_CLUSTER_ZONE="us-central1-a"
      ;;
```

The indentation is off - please fix

File: .kokoro/psm_interop_kokoro_lib.sh [hunk 3]
```diff
@@ -694,6 +739,10 @@ activate_gke_cluster() {
       GKE_CLUSTER_NAME="interop-test-psm-basic"
       GKE_CLUSTER_ZONE="us-central1-c"
       ;;
+    GKE_CLUSTER_DUAL_STACK)
+          GKE_CLUSTER_NAME="psm-interop-lb-primary-ds"
```
Comment (sergiitk) [inline .kokoro/psm_interop_kokoro_lib.sh:743]: Would be great if we used the new cluster name here
```suggestion
          GKE_CLUSTER_NAME="psm-interop-dualstack"
```

File: .kokoro/psm_interop_kokoro_lib.sh [hunk 4]
```diff
@@ -694,6 +739,10 @@ activate_gke_cluster() {
       GKE_CLUSTER_NAME="interop-test-psm-basic"
       GKE_CLUSTER_ZONE="us-central1-c"
       ;;
+    GKE_CLUSTER_DUAL_STACK)
```
Comment (sergiitk) [inline .kokoro/psm_interop_kokoro_lib.sh:742]: nit
```suggestion
    GKE_CLUSTER_DUALSTACK)
```

File: .kokoro/psm_interop_kokoro_lib.sh [hunk 5]
```diff
@@ -158,6 +158,51 @@ psm::security::run_test() {
...
+}
+
+#######################################
+# Prepares the list of tests in DualStack test suite.
+# Globals:
+#   TESTS: Populated with tests in PSM DualStack test suite.
+#######################################
+psm::dual_stack::get_tests() {
+  TESTS=(
+    "test_dual_stack"
```
Comment (sergiitk) [inline .kokoro/psm_interop_kokoro_lib.sh:179]:
```suggestion
    "test_dualstack"
```

File: .kokoro/psm_interop_kokoro_lib.sh [hunk 6]
```diff
@@ -158,6 +158,51 @@ psm::security::run_test() {
...
+#   Writes the output of test execution to stdout, stderr
+#   Test xUnit report to ${TEST_XML_OUTPUT_DIR}/${test_name}/sponge_log.xml
+#######################################
+psm::dual_stack::run_test() {
+  local test_name="${1:?${FUNCNAME[0]} missing the test name argument}"
+
+  # Only java supports extra checks for certificate matches (via channelz socket info).
+  if [[ "${GRPC_LANGUAGE}" != "java"  ]]; then
+    PSM_TEST_FLAGS+=("--nocheck_local_certs")
+  fi
```
Comment (sergiitk) [inline .kokoro/psm_interop_kokoro_lib.sh:200]: This is only needed for security tests

```suggestion
```

File: .kokoro/psm_interop_kokoro_lib.sh [hunk 7]
```diff
@@ -158,6 +158,51 @@ psm::security::run_test() {
...
+
```
Comment (sergiitk) [inline .kokoro/psm_interop_kokoro_lib.sh:201]:
```suggestion
  PSM_TEST_FLAGS+=(
    "--flagfile=config/common-dualstack.cfg"
  )
```

Let's create config/common-dualstack.cfg, with the following content:

```
# Common config file for PSM DualStack tests.
--resource_prefix=psm-ds
--enable_dualstack
--noenable_workload_identity
```

File: .kokoro/psm_interop_kokoro_lib.sh [hunk 8]
```diff
@@ -158,6 +158,51 @@ psm::security::run_test() {
...
 }
 
+# --- DualStack TESTS ------------------
+
+#######################################
+# DualStack Test Suite setup.
+# Outputs:
+#   Prints activated cluster names.
+#######################################
+psm::dual_stack::setup() {
```
Comment (sergiitk) [inline .kokoro/psm_interop_kokoro_lib.sh:168]: Here and elsewhere: rename `dual_stack` to `dualstack`.
This is to be consistent with how we already refer to this functionality in the framework (e.g. `--enable_dualstack` flag).

**Review by sergiitk — CHANGES_REQUESTED** [review]

**Comment by aranjans** [comment]
@sergiitk thAnks for your review, I have updated the PR with all the suggestions.
fyi: I have updated the GKE cluster name but @purnesh42H is yet to create the new cluster with same name.

**Review by sergiitk — APPROVED** [review]

**Pull request merged**